### ISSUE TITLE
The question about localSlidingWindowIntersectThreshold and localSlidingWindowEuclideanThreshold

### ISSUE THREAD

**Issue by xiamenwcy** [Trojahn/FAST#1]
Hi,Thank your for this very beautiful work! But when I read source code,I find a question that In class ShotSegmentation ,member variable like localSlidingWindowIntersectThreshold and localSlidingWindowEuclideanThreshold has not been used.I guss you are writing wrong in function ShotSegmentation::segment()  , ` distIntersec[i] <= this->swIntersectThreshold ` should  corrected to be   `distIntersec[i] <= this->localSlidingWindowIntersectThreshold` as well as `distEuclidean[i] >= this->swEuclideanThreshold ` should also corrected to be `distEuclidean[i] >= this->localSlidingWindowEuclideanThreshold `   Do you agree with me?

**Comment by Trojahn** [comment]
Hi, thank you for your report! I hope that this tool may be useful to you! :)

For the error mentioned, let my try to explain what I desired of this particular piece of software and what i found out.

The objective of the for-loop mentioned is to find the "hard cuts" of the shot segmentation. It's those that have a way too large one-on-one dissimilarity between them and almost always are a shot transition (or, at least, it should be in the histogram analysis). That bunch of conditions try to identify just that.

These "hard cuts" points, found by using the corresponding thresholds, identifies the "sliding window" positions. That is the desired behavior, as also implemented in the FAST main function:
`ss.setGradualThreshold(gradualHeuristic);
ss.setSlidingWindowsIntersect(swIntersection);
ss.setSlidingWindowsEuclidean(swEuclidean);
ss.setLocalSlidingWindowIntersect(localSWIntersection);
ss.setLocalSlidingWindowEuclidean(localSWEuclidean);`

The thresholds mentioned, however, was also incorrectly used in the `calcThresholdEuclidean` and `calcThresholdIntersection` methods. That's indeed a mistake as it where I indeed intended to use the mentioned thresholds.

See the updated code (going live ASAP) and keep in touch if you don't agree or find another mistake in the source... I think it should make more sense now.

Sorry for the mess :+1:
